### DISCUSSION TITLE
If server times out...

### DISCUSSION THREAD

**Discussion by nostriel** [hazzelnuts/ntfy-for-delphi#38] [Q&A]
Hi.
I found about this app watching Delphicon 2023. It's great as is easy to implement.
First, I wanted to run Ntfy.Publish async, so I run it  inside TTask.Run(); (I don't want the user to wait for the app to respond)
But I want to catch this situation: 
The server is offline. Indy has by default a 20 second timeout, and only after this it throws the timeout error exception but I stay less than 20 seconds in the app.
This will force me to implement a wait system for the thread to finish, otherwise it will crash.
Do you think is a good ideea as to expose FIdHttp.ConnectTimeout to set custom timeouts?
Another solution might be exposing the Disconnect function of FIdHttp, so on exit I would disconnect forcefully.

**Comment by p-samuel** [comment]
Hi @nostriel, thanks for the feedback. For me both options are okay as it gives more flexibility. The only issue is where this should be implemented. I don't know if you noticed but there is a public interface called `Core `and another one called `Api`. This `Api `wraps the routines belonging to the communication components (Indy and NetHTTP). If I was about to make this small enhancement I would start looking on this public interface, and then  implement it on the interfaced classes. My only concern it is if access to this method should be transferred to the `Core `interface or if we could expose the `Api `interface in the `Core `interface. Do you understand my point?

I can take a look on that as soon as I finish the other two PR that are pending for I am lacking a bit of time at this moment, but if you wish you can make a fork of this repository, implement these changes by yourself and then send a PR to this repo again as a form to you contribute with the project.
